### PR TITLE
feat: add platform token parameter

### PR DESCRIPTION
**Pull request by huskar-t**
TD-15079

### CODE REVIEW

**Review by huolibo — APPROVED** [review]
LGTM

**Review by zitsen — APPROVED** [review]
LGTM

**Review by sangshuduo — CHANGES_REQUESTED** [review]
see my comment

**Comment by huskar-t** [comment]
after readjustment

**Review by huolibo — APPROVED** [review]
LGTM

File: taosRestful/connection.go [hunk 1]
```diff
@@ -56,16 +62,47 @@ func newTaosConn(cfg *config) *taosConn {
...
-		User:   url.UserPassword(cfg.user, cfg.passwd),
 		Host:   fmt.Sprintf("%s:%d", cfg.addr, cfg.port),
 		Path:   path,
 	}
-	basic := "Basic " + base64.StdEncoding.EncodeToString([]byte(cfg.user+":"+cfg.passwd))
-	tc.header = map[string][]string{"Authorization": {basic}}
+	if cfg.token != "" {
+		tc.url.RawQuery = fmt.Sprintf("token=%s", cfg.token)
+	}
+	//loginUrl := &url.URL{
```
Comment (sangshuduo) [inline taosRestful/connection.go:71]: we should avoid keeping too many lines of unused code in the release branch.

**Review by sangshuduo — APPROVED** [review]
LGTM with a comment

**Pull request merged**